applications/FemToDemApplication/custom_processes/expand_wet_nodes_process.cpp

```diff
@@ -92,4 +97,58 @@ void ExpandWetNodesProcess::ExpandWetNodes(
 /***********************************************************************************/
 /***********************************************************************************/
 
-}
\ No newline at end of file
+void ExpandWetNodesProcess::ExpandWetNodesIfTheyAreSkin()
+{
+    Parameters skin_process_parameters = Parameters(R"(
+    {
+        "name_auxiliar_model_part"              : "SkinModelPart",
+        "name_auxiliar_condition"               : "Condition",
+        "list_model_parts_to_assign_conditions" : [],
+        "echo_level"                            : 0
+    })");
+
+    auto skin_process = SkinDetectionProcess<2>(mrModelPart, skin_process_parameters); // TODO what's up in 3d?
```

Comment:
You can check the dimension with DOMAIN_SIZE

Reply:
auto& r_process_info = mrModelPart.GetProcessInfo();
    const int dimension = r_process_info[DOMAIN_SIZE];
    auto skin_process = SkinDetectionProcess<dimension>(mrModelPart, skin_process_parameters);

This not compiles... Shall I do some If dim == 2 {SkinDetectionProcess<2>(mrModelPart, skin_process_parameters);}
else...
?

Reply:
Yes because the templates are evaluated on compilation time

Reply:
Solved